### ISSUE TITLE
Switch to a new resolution without buffering in the video player.

### ISSUE THREAD

**Issue by dkanada** [TeamNewPipe/NewPipe#659]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.

It would be nice if the video player didn't switch to the new resolution until it built up a buffer. This would allow the video to continue playing rather than stop to reload the video.

**Comment by turboteckel** [comment]
Isn't this a YouTube thing?

**Comment by ildar** [comment]
Yes, smooth resolution switch is YT feature, I guess not so trivial.

**Comment by theScrabi** [comment]
Well with exoplayer we could do that. But we needed someone who knows how to handle exoplayer well.

**Comment by ildar** [comment]
The man appeared?

**Comment by theScrabi** [comment]
They come and go. But no this has not happen yet :/

**Comment by ildar** [comment]
@avently , can you give your opinion on this issue? Thanks!

**Comment by avently** [comment]
@ildar when a video from playlist ends I don't see a loading indicator. This means somehow exoplayer can preload a video. But as you know this thing doesn't work when you skip a video manually with a right arrow button. Theoretically it is possible to switch to a next video (or another resolution) without a gap but practically I couldn't find a place in code where we can tell exoplayer to start preloading from a specified playback position. Maybe someone has a time to look at the source of Exoplayer and find such a place of code where magic happens

**Comment by Stypox** [comment]
@opusforlife2 Idk, it seems like a too-specific label, a good title would do just as well
